Classes/Middleware/Frontend/LanguageModeSwitch.php

```diff
@@ -138,4 +155,27 @@ private function getNewLanguageWithMode(
             ])
         );
     }
+
+    private function pageHasStandAloneContent(int $pageId, int $languageId): bool
+    {
+        $queryBuilder = clone $this->queryBuilder;
```

Comment:
Ich nutze in der Klasse den gleichen QueryBuilder für requests auf 2 verschiedene Tabellen.

Die reine Lehre ist das nicht, da für jede Tabelle ein eigener QueryBuilder abgefragt werden müsste.
Wir machen das bei uns im Projekt so für Tabellen, die sowieso eng gekoppelt sind (Es gibt im Core viele Abfragen mit JOINs über pages und tt_content hinweg) und die deshalb nie auf verschiedenen Connections sein können.

Wenn das so nicht gewünscht ist, kann ich aber auch sauber einen zweiten QueryBuilder injecten.